pkg/controller/controller_ref_manager.go

```diff
@@ -113,7 +113,7 @@ func (m *PodControllerRefManager) AdoptPod(pod *v1.Pod) error {
 			strings.Join([]string{pod.Namespace, pod.Name, string(pod.UID)}, "_"))
 	}
 	addControllerPatch := fmt.Sprintf(
-		`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true}],"uid":"%s"}}`,
+		`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"uid":"%s"}}`,
```

Comment:
Existing controllers won't be blocked by their dependents during a foreground GC, unless the user manually patch the ownerReferences.

Doing a migration will be nice for users, or we can just communicate the fact via docs.